sycl/source/detail/platform_impl.cpp

```diff
@@ -49,6 +50,109 @@ platform_impl_host::get_devices(info::device_type dev_type) const {
   return res;
 }
 
+struct DevDescT {
+  const char *devName = nullptr;
+  int devNameSize = 0;
+
+  const char *minDriverVer = nullptr;
+  int minDriverVerSize = 0;
+};
+
+static std::vector<DevDescT> getWhiteListDesc() {
+  // TODO: Replace with const char *str =
+  // SYCLConfig<SYCL_DEVICE_WHITE_LIST>::get();
+  const char *str = getenv("SYCL_DEVICE_WHITE_LIST");
+  if (!str)
+    return {};
+
+  std::vector<DevDescT> decDescs;
+  const char devNameStr[] = "DeviceName";
+  const char driverVerStr[] = "DriverVersion";
+  decDescs.emplace_back();
+  while (*str != '\0') {
+    const char **dstPtr = nullptr;
+    int *Size = nullptr;
+
+    // -1 to avoid comparing null terminator
+    // TODO: Change devNameStr so it doesn't contain null terminator
+    if (0 == strncmp(devNameStr, str, sizeof(devNameStr) - 1)) {
+      dstPtr = &decDescs.back().devName;
+      Size = &decDescs.back().devNameSize;
+      str += sizeof(devNameStr);
+    } else if (0 == strncmp(driverVerStr, str, sizeof(driverVerStr) - 1)) {
+      dstPtr = &decDescs.back().minDriverVer;
+      Size = &decDescs.back().minDriverVerSize;
+      str += sizeof(driverVerStr);
+    }
+
+    if ('\0' == *str || ',' == *str || !dstPtr) {
+      // Malformed white list.
+      // TODO: throw an exception here.
```

Comment:
Please resolve this TODO, it shouldn't be so hard.

However, I'm not sure what RT should do in general, if white list is specified, but malformed: do it need to silently continue or throw an exception?

Reply:
Yeah, it's not hard. In order to do that we should answer the question you have asked.
I prefer just throwing an exception as it should be clear to final user that he does something wrong. Otherwise he silently gets unexpected results.

Reply:
Throwing exceptions on unexpected symbols now.